rfcs/convergence/fixed-versions-for-prerelease.md

```diff
@@ -0,0 +1,288 @@
+guarantee that APIs won't break, especially when a component has not even been used in any realistic
+scenario by partners.
+
+Realistically for the Fluent team, bugs found during either `alpha` or `beta` phases must be addressed or fixed
+upon release. If a bug or an API design does not match customer requirements in either phases, we **are entitled** break the
+API in a further iteration to improve.
+
+#### Pros
+
+- No extra work required, just a convention to release all new components straight to beta
+- Avoid dependency hell by recommending customers to only use the suite package
+- No extra effort for customers other than install the correct suite package version
+
+#### Cons
+
+- Not sure what customer expectations are for new components in beta
+- Need to properly define alpha vs beta -> what are our requirements ?
+
+### Provide resolutions to customers
+
+If we must release a component into the `alpha`, consumers can still apply manual resolutions.
+To support this scenario, we can provide a generated set of yarn resolutions to allow all Fluent dependencies to
+work together from the a consuming app.
+
+The generated resolutions must be for a specific version of the `react-components` package.
+
+We can provide this functionality either on our docsite or as a published CLI package.
+
+#### Pros
+
+- Easier debugging for partners, there should only be one Fluent UI version in node_modules
+
+#### Cons
+
+- Need to work on a project to generate resolutions
+- Temporary, no need for this after major release
+
+### Lockstep versioning for `beta` components
+
+Lockstep versioning does not solve the problem, but makes finding compatible versions of `beta` and `alpha` packages
+easier. Using the below example:
+
+- `@fluentui/react-components` -> 9.0.0.beta.13
+
+  - `@fluentui/react-shared-contexts` -> 9.0.0.beta.13
+  - `@fluentui/react-make-styles` -> 9.0.0.beta.3
+
+- `@fluentui/react-dropdown` -> 9.0.0.alpha.1
+  - `@fluentui/react-shared-contexts` -> 9.0.0.beta.10
+  - `@fluentui/react-make-styles` -> 9.0.0.beta.10
+
+It's clear that we only need to find any version of `react-dropdown` that uses `beta.10` dependencies. Any
+future version mismatch can be easily investigated and fixed.
+
+For the problem of fixed versions, lockstep only solves a temporary problem since
+on major release Fluent can use full semver features.
+
+#### Pros
+
+- Easy to find compatible versions for alpha packages
+- Easier debugging for partners, there should only be one Fluent UI version in node_modules
+- Don't have to get rid of lockstep after major release
+
+#### Cons
+
+- Need infra to support lockstep versioning
+- Doesn't solve the dependency issue, just makes it easier to manage
+
+### Deep imports for `alpha` packages
+
+> Demo here: https://github.com/ling1726/multiple-entrypoints-test
+
+This solution reintroduces deep imports with the path `@fluentui/react-components/lib/alpha`. The only workaround that
```

Comment:
```suggestion
This solution reintroduces deep imports with the path `@fluentui/react-components/alpha`. The only workaround that
```

I guess that it should be without `/lib`, right?
As an option, we can consider to have `unstable` instead of `/alpha`.

Reply:
I don' t think in the long term we should have this kind of import for consuming unstable changes. On major version release, we should be able to take advantage of semver to let customers consume unstable component packages

Reply:
Opted to use `unstable` based on extra discussion what we actually want out of this